compiler/hash-ast/src/module.rs

```diff
@@ -34,27 +35,32 @@ impl<'c> ModuleBuilder<'c> {
         Self::default()
     }
 
-    pub(crate) fn add_module_at(
+    pub fn add_module_at(
```

Comment:
It was due to the fact that once the `add_module_at` is called, the contents of the `source` are moved and later accesses would require copying. This was a hack to get the reporting stuff to work (since it needs both the module index and the source available). We should take a look at fixing up this hack.

Reply:
Actually this was pretty easy to fix.